plugins/alertSound/alertSound.pl

```diff
@@ -1,162 +1,149 @@
+		my $display = ($::monsters_lut{$args->{type}} ne "")
+			? $::monsters_lut{$args->{type}}
+			: "Unknown ".$args->{type};
+		alertSound("monster $display");
+	}
+}
+sub player {
+# eventList player <player name>
+# eventlist player *
+# eventList GM near
+	my (undef, $args) = @_;
+	my $name = $args->{player}{name};
+	
+	for (my $i = 0; exists $config{"alertSound_".$i."_eventList"}; $i++) {
+		next if (!$config{"alertSound_".$i."_eventList"});
+		if (Utils::existsInList($config{"alertSound_".$i."_eventList"}, "player *")) {
+			alertSound("player *");
+			return;
+		}
+	}
+
+	if ($name =~ /^([a-z]?ro)?-?(Sub)?-?\[?GM\]?/i) {
+		alertSound("GM near");
+	} else {
+		alertSound("player $name");
+	}
+}
+sub private {
+# eventList private GM chat
+# eventList private chat
+	my (undef, $args) = @_;
+	if ($args->{privMsgUser} =~ /^([a-z]?ro)?-?(Sub)?-?\[?GM\]?/i) {
+		alertSound("private GM chat");
+	} else {
+		alertSound("private chat");
+	}
+}
+sub public {
+# eventList public GM chat
+# eventList npc chat
+# eventList public chat
+	my (undef, $args) = @_;
+	if ($args->{pubMsgUser} =~ /^([a-z]?ro)?-?(Sub)?-?\[?GM\]?/i) {
+		alertSound("public GM chat");
+	} elsif (unpack("V", $args->{pubID}) == 0) {
+		alertSound("npc chat");
+	} else {
+		alertSound("public chat");
+	}
+}
+sub system_message {
+# eventList system message
+	alertSound("system message");
 }
 
-
-##
-# alertSound($event)
-# $event: unique event name
-#
-# Plays a sound if alertSound is enabled,
-# and if a sound is specified for the event.
-#
-# The config option "alertSound_#_eventList" should have a comma
-# seperated list of all the desired events.
-#
-# Supported events:
-# public chat, public GM chat, private chat, private GM chat, emoticon, system message
-# map change, GM near, monster <monster name>
 sub alertSound {
-        return if (!$config{'alertSound'});
-        my $event = shift;
-        my $i = 0;
-        for (my $i = 0; exists $config{"alertSound_".$i."_eventList"}; $i++) {
-                next if (!$config{"alertSound_".$i."_eventList"});
-                if (Utils::existsInList($config{"alertSound_".$i."_eventList"}, $event)
-                && (!$config{"alertSound_".$i."_notInTown"} || !$cities_lut{$field->baseName().'.rsw'})
-                && (!$config{"alertSound_".$i."_inLockOnly"} || $field->baseName() eq $config{'lockMap'})) {
-                        message "Sound alert: $event\n", "alertSound";
-                        Utils::Win32::playSound($config{"alertSound_".$i."_play"});
-                        return;
-                }
-        }
+	my $event = shift;
+	for (my $i = 0; exists $config{"alertSound_".$i."_eventList"}; $i++) {
+		next if (!$config{"alertSound_".$i."_eventList"});
+		if (Utils::existsInList($config{"alertSound_".$i."_eventList"}, $event)
+			&& (!$config{"alertSound_".$i."_notInTown"} || !$cities_lut{$field->baseName().'.rsw'})
+			&& (!$config{"alertSound_".$i."_inLockOnly"} || $field->baseName() eq $config{'lockMap'})) {
```

Comment:
instead of putting those two values here (`notInTown` and `inLockOnly`) i think it would be better to insert a `Misc::checkSelfCondition` here, to check these two and a lot more